tldr/dfg_extractor.py

```
@@ -1506,6 +1699,83 @@ def search(node):
+        )
+
+    # Parse with tree-sitter
+    r_lang = Language(tree_sitter_r.language())
+    parser = Parser(r_lang)
+    source_bytes = code.encode("utf-8")
+    tree = parser.parse(source_bytes)
+
+    # Find the function (R functions: name <- function(args) { ... })
+    func_node = None
+
+    def find_r_function(node) -> Any:
```

Ruff (0.14.13)

Dynamically typed expressions (typing.Any) are disallowed in `find_r_function`

(ANN401)